processing/src/main/java/io/druid/query/aggregation/LastAggregatorFactory.java

```diff
@@ -0,0 +1,335 @@
+              }
+            }
+          };
+        } else if (value.equals(TYPE_DOUBLE)) {
+          return new DoubleLastBufferAggregator(null, null)
+          {
+            @Override
+            public void aggregate(ByteBuffer buf, int position)
+            {
+              SerializablePair<Long, Double> pair = (SerializablePair<Long, Double>) selector.get();
+              long lastTime = buf.getLong(position);
+              if (pair.lhs >= lastTime) {
+                buf.putLong(position, pair.lhs);
+                buf.putDouble(position + Longs.BYTES, pair.rhs);
+              }
+            }
+          };
+        }
+        throw new IAE("undefined type");
+      }
+    };
+  }
+
+  @Override
+  public AggregatorFactory getMergingFactory(AggregatorFactory other) throws AggregatorFactoryNotMergeableException
+  {
+    if (other.getName().equals(this.getName()) && this.getClass() == other.getClass() &&
+        other.getTypeName().equals(this.getTypeName())) {
+      return getCombiningFactory();
+    } else {
+      throw new AggregatorFactoryNotMergeableException(this, other);
+    }
+  }
+
+  @Override
+  public List<AggregatorFactory> getRequiredColumns()
+  {
+    return Arrays.<AggregatorFactory>asList(new LastAggregatorFactory(fieldName, fieldName, value));
+  }
+
+  @Override
+  public Object deserialize(Object object)
+  {
+    Map map = (Map) object;
+    if (value.equals(TYPE_LONG)) {
+      return new SerializablePair<>(((Number) map.get("lhs")).longValue(), ((Number) map.get("rhs")).longValue());
+    } else if (value.equals(TYPE_DOUBLE)) {
+      return new SerializablePair<>(((Number) map.get("lhs")).longValue(), ((Number) map.get("rhs")).doubleValue());
+    }
+    throw new IAE("undefined type");
+  }
+
+  @Override
+  public Object finalizeComputation(Object object)
+  {
+    return ((SerializablePair<Long, Object>) object).rhs;
+  }
+
+  @Override
+  @JsonProperty
+  public String getName()
+  {
+    return name;
+  }
+
+  @JsonProperty
+  public String getFieldName()
+  {
+    return fieldName;
+  }
+
+  @JsonProperty
+  public String getValue()
+  {
+    return value;
+  }
+
+  @Override
+  public List<String> requiredFields()
+  {
+    return Arrays.asList(Column.TIME_COLUMN_NAME, fieldName);
+  }
+
+  @Override
+  public byte[] getCacheKey()
+  {
+    byte[] fieldNameBytes = StringUtils.toUtf8(fieldName);
+
+    return ByteBuffer.allocate(1 + fieldNameBytes.length).put(CACHE_TYPE_ID).put(fieldNameBytes).array();
+  }
+
+  @Override
+  public String getTypeName()
+  {
+    return value.equals(TYPE_DOUBLE) ? "float" : value;
+  }
+
+  @Override
+  public int getMaxIntermediateSize()
+  {
+    if (value.equals(TYPE_LONG)) {
+      return Longs.BYTES * 2;
+    } else if (value.equals(TYPE_DOUBLE)) {
+      return Longs.BYTES + Doubles.BYTES;
+    }
+    throw new IAE("undefined type");
+  }
+
+  @Override
+  public Object getAggregatorStartValue()
```

Comment:
It's ok to throw UnsupportedOperationException here, nothing calls this and we should probably get rid of it at some point.